packages/orchestrator-ui-components/src/components/WFOWorkflowSteps/WFOWorkflowStepList/WFOWorkflowStepList.tsx

```diff
@@ -20,36 +20,53 @@ export const WFOWorkflowStepList = React.forwardRef(
 
         const t = useTranslations('processes.steps');
 
-        const allDetailsClosedState = new Map(
-            steps.map((_, index) => [index, false]),
-        );
-        const allDetailsOpenedState = new Map(
-            steps.map((_, index) => [index, true]),
+        const initialStepListItems: StepListItem[] = steps.map((step) => ({
+            step,
+            isExpanded: false,
+        }));
+        const [stepListItems, setStepListItems] =
+            useState(initialStepListItems);
+
+        const allStepsAreExpanded = stepListItems.every(
+            (item) => item.isExpanded,
         );
 
-        const [stepDetailStates, setStepDetailStates] = useState<
-            Map<number, boolean>
-        >(new Map(allDetailsClosedState));
+        const setExpandedStateStepListItems = (isExpanded: boolean) => {
+            setStepListItems(
+                stepListItems.map((item) => ({
+                    ...item,
+                    isExpanded,
+                })),
+            );
+        };
 
-        const toggleAllDetailsIsOpen = () => {
-            if (Array.from(stepDetailStates).every((s) => s[1])) {
-                setStepDetailStates(new Map(allDetailsClosedState));
-            } else {
-                setStepDetailStates(new Map(allDetailsOpenedState));
-            }
+        const handleToggleExpandedStepListItem = (
```

Comment:
Same here. I would argue that you can toggle and expandedStepListItem, you can collapse it since if you toggle it back the function doesn't do what is called anymore. So just toggleStepListItem?

Reply:
todo: try to make in line with "setExpandedStateStepListItems"

Reply:
Done